src/Component/Carousel.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Class Carousel.
+ *
+ * @package Amp\AmpWP
+ */
+
+namespace Amp\AmpWP\Component;
+
+use DOMDocument;
+use DOMElement;
+use AMP_DOM_Utils;
+
+/**
+ * Class Carousel
+ *
+ * Gets the markup for an <amp-carousel>.
+ *
+ * @internal
+ * @since 1.5.0
+ */
+class Carousel {
```

Comment:
As we marked the `Carousel` as `@internal`, we should mark the `Carousel` as `final` and use `private` accessibility for the properties.

In general, final+private should be the default, unless we explicitly want something to be extendable, not the other way around.

We can always turn a final class into an extendable class. The other way it not so easy because third-party code might already be relying on it being extendable.

```suggestion
final class Carousel {
```